unified_planning/model/state.py

```diff
@@ -13,12 +13,91 @@
 # limitations under the License.
 #
 
+from typing import Dict, List, Optional
 import unified_planning as up
+from unified_planning.exceptions import UPUsageError
 
 
-class State:
-    '''This is an abstract class representing a classical state'''
+class ROState:
+    '''This is an abstract class representing a classical Read Only state'''
 
-    def get_value(self, f: 'up.model.FNode'):
+    def get_value(self, value: 'up.model.FNode') -> 'up.model.FNode':
         '''Returns the value assignment for the given fluent expression'''
         raise NotImplementedError
+
+
+class RWState(ROState):
+    '''This is an abstract class representing a classical Read/Write state'''
+
+    def set_values(self, new_values: Dict['up.model.FNode', 'up.model.FNode']) -> 'RWState':
+        '''Returns a new RWState, where every value assignment not in new_values.keys() remains the same
+        and every other value assignment is updated following the new_values map.'''
+        raise NotImplementedError
+
+
+class UPRWState(RWState):
+    # TODO: DOCUMENT
+    max_ancestors: int = 3
```

Comment:
Document this filed. Moreover use UPPER_CASE for constants. Finally, 3 is a very low value, I would go with `20`